aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SyncMetricsObserver.java

```diff
@@ -0,0 +1,135 @@
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.datastore.syncengine;
+
+import com.amplifyframework.core.Amplify;
+import com.amplifyframework.core.async.Cancelable;
+import com.amplifyframework.core.model.Model;
+import com.amplifyframework.datastore.DataStoreChannelEventName;
+import com.amplifyframework.datastore.DataStoreConfigurationProvider;
+import com.amplifyframework.datastore.DataStoreException;
+import com.amplifyframework.datastore.DataStoreItemChange;
+import com.amplifyframework.datastore.events.ModelSyncedEvent;
+import com.amplifyframework.datastore.storage.LocalStorageAdapter;
+import com.amplifyframework.datastore.storage.StorageItemChange;
+import com.amplifyframework.hub.HubChannel;
+import com.amplifyframework.hub.HubEvent;
+import com.amplifyframework.logging.Logger;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Class that encapsulates the logic of listening to local DataStore changes
+ * and emitting metrics when base/delta sync completes.
+ */
+class SyncMetricsObserver {
+    private static final Logger LOG = Amplify.Logging.forNamespace("amplify:aws-datastore");
+    private Map<String, ModelSyncMetrics> metricsByModel;
+    private final Cancelable itemChangeObserver;
+    private final List<Class<? extends Model>> syncableModels;
+
+    /**
+     * Constructor that sets up an observer to watch for mutations
+     * made by the sync process on the local DataStore.
+     * @param localStorageAdapter A reference to the implementation of the {@link LocalStorageAdapter}.
+     * @param syncableModels A list of all the syncable models.
+     * @param dataStoreConfigurationProvider A reference to the DataStore configuration.
+     */
+    SyncMetricsObserver(LocalStorageAdapter localStorageAdapter,
+                        List<Class<? extends Model>> syncableModels,
+                        DataStoreConfigurationProvider dataStoreConfigurationProvider) {
+        itemChangeObserver = localStorageAdapter.observe(
+            this::onItemChange,
+            this::onError,
+            this::onComplete
+        );
+        metricsByModel = new ConcurrentHashMap<>();
+        this.syncableModels = syncableModels;
+    }
+
+    private void onComplete() {
+        LOG.debug("Initial sync metrics observer completed.");
+        itemChangeObserver.cancel();
+    }
+
+    private void onError(DataStoreException dataStoreException) {
+        LOG.warn("Unable to gather metrics for remote sync.", dataStoreException);
+    }
+
+    private void onItemChange(StorageItemChange<? extends Model> itemChange) {
+        boolean isSyncProcessChange = StorageItemChange.Initiator.SYNC_ENGINE.equals(itemChange.initiator());
+        if (isSyncProcessChange) {
```

Comment:
Maybe invert this, so you can save 4 columns for the main body of the method.
e.g.
```java
if (!isSyncProcessChange) {
    return;
}
ahaSweetISavedSomeCols();
```

Reply:
No longer needed.